tests/unit/helpers/courseBills.test.js

```diff
@@ -672,103 +674,180 @@ describe('list', () => {
+    findOneCourse.returns(SinonMongoose.stubChainedQueries(course, ['lean']));
+
+    await CourseBillHelper.createBillList(payload);
+
+    sinon.assert.calledOnceWithExactly(
+      createCourseBill,
+      {
+        course: payload.course,
+        mainFee: {
+          price: 100,
+          count: 1,
+          countUnit: GROUP,
+          description: 'test',
+        },
+        companies: payload.companies,
+        payer: payload.payer,
+        maturityDate: payload.maturityDate,
+      });
     sinon.assert.notCalled(addBillingPurchase);
+    sinon.assert.notCalled(insertManyCourseBills);
   });
 
-  it('should create a course bill (with percentage)', async () => {
-    const courseId = new ObjectId();
+  it('should create one bill with trainer fees for INTRA course', async () => {
     const companyId = new ObjectId();
     const course = {
-      _id: courseId,
-      prices: [{ company: companyId, global: 1200 }],
+      _id: new ObjectId(),
+      type: INTRA,
+      prices: [{ company: companyId, global: 1200, trainerFees: 200 }],
     };
+    const billCreated = { _id: new ObjectId() };
     const payload = {
-      course: courseId,
-      companies: [companyId, new ObjectId()],
-      mainFee: { price: 120, count: 1, percentage: 10 },
+      course: course._id,
+      quantity: 1,
+      mainFee: { price: 100, count: 1, countUnit: GROUP, percentage: 10 },
+      companies: [companyId],
       payer: { fundingOrganisation: new ObjectId() },
-      maturityDate: '2022-03-08T00:00:00.000Z',
+      maturityDate: '2025-04-29T22:00:00.000Z',
     };
 
     findOneCourse.returns(SinonMongoose.stubChainedQueries(course, ['lean']));
-    await CourseBillHelper.create(payload);
+    createCourseBill.returns(billCreated);
 
-    sinon.assert.calledOnceWithExactly(create, payload);
-    sinon.assert.notCalled(addBillingPurchase);
-    SinonMongoose.calledOnceWithExactly(
-      findOneCourse,
-      [
-        { query: 'findOne', args: [{ _id: payload.course }, { prices: 1 }] },
-        { query: 'lean', args: [] },
-      ]
+    await CourseBillHelper.createBillList(payload);
+
+    sinon.assert.calledOnceWithExactly(
+      createCourseBill,
+      {
+        course: payload.course,
+        mainFee: {
+          price: 100,
+          count: 1,
+          countUnit: GROUP,
+          percentage: 10,
+        },
+        companies: payload.companies,
+        payer: payload.payer,
+        maturityDate: payload.maturityDate,
+      });
+    sinon.assert.calledOnceWithExactly(
+      addBillingPurchase,
+      billCreated._id,
+      {
+        price: 20,
+        count: 1,
+        percentage: 10,
+        billingItem: TRAINER_FEES_BILLING_ITEM,
+      }
     );
+    sinon.assert.notCalled(insertManyCourseBills);
   });
 
-  it('should create a course bill (with percentage and trainer fees)', async () => {
-    const courseId = new ObjectId();
+  it('should create a course bill with global for INTRA course', async () => {
```

Comment:
Plutot que de preciser `with global`, j'aurais mis `with percentage and wihtout trainerFees`